builder/builder.go

```diff
@@ -852,3 +875,66 @@ func parseUint32(s string) (uint32, error) {
 	}
 	return uint32(parsed), nil
 }
+
+func createCompressedArchive(dst string, srcs ...string) (err error) {
+	defer func() {
+		if err != nil {
+			err = errors.Wrapf(err, "couldn't create compressed archive %s", dst)
+		}
+	}()
+
+	headers := make([]*tar.Header, len(srcs))
+	for i, src := range srcs {
+		var fi os.FileInfo
+		fi, err = os.Lstat(src)
+		if err != nil {
+			return err
+		}
+		if !fi.Mode().IsRegular() {
+			return errors.Errorf("%s has unsupported type of file", src)
+		}
+		headers[i], err = tar.FileInfoHeader(fi, "")
+		if err != nil {
+			return err
+		}
+	}
+
+	archive := &bytes.Buffer{}
+
+	xw, err := swupd.NewExternalWriter(archive, "xz")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if xw != nil {
+			_ = xw.Close()
+		}
+	}()
+
+	tw := tar.NewWriter(xw)
+	for i, hdr := range headers {
+		err = tw.WriteHeader(hdr)
+		if err != nil {
+			return err
+		}
+		var srcData []byte
+		srcData, err = ioutil.ReadFile(srcs[i])
```

Comment:
Probably ought to think if we want to read large amounts of data into memory, just to pass it on to tar or if it makes sense to read it in chunks, e.g. 256KB. Something for the future, not now.

Reply:
This function job was to archive two small regular files (Manifest.MoM and Manifest.MoM.sig), so I opted for the tradeoff of doing all in memory so I will only write a file when it is good (makes error handling easier as I'll never need to care about invalid file being written).

I'll clarify that scope in the function comment.

Reply:
@icarus-sparry could you please take another look?